analysis_driver/dataset.py

```diff
@@ -351,63 +353,55 @@ def rapid_samples_by_lane(self):
-            flowcell = set(self.lims_run.parent_processes()).pop().output_containers()[0]
-            for lane, artifact in flowcell.placements.items():
-                if len(artifact.reagent_labels) > 1:
+            for lane in self.run_status['lanes']:
+                if len(lane['samples']) > 1:
                     continue  # we don't want to run rapid processing on pools
+                sample_data = lane['samples'][0]
 
-                assert len(artifact.samples) == 1
-                sample = artifact.samples[0]
-                if sample.udf.get('Rapid Analysis') == 'Yes':
-                    s = sample.udf.copy()
-                    s['sample_id'] = sample.name
-                    s['project_id'] = sample.project.name
-                    self._rapid_samples_by_lane[lane.split(':')[0]] = s
+                sample_info = rest_communication.get_document('lims/sample_info',
+                                                              match={'sample_id': sample_data['sample_id']})
+                if sample_info.get('Rapid Analysis') == 'Yes':
+                    self._rapid_samples_by_lane[str(lane['lane'])] = sample_info
 
         return self._rapid_samples_by_lane
 
-    def _run_elements_from_lims(self):
+    @property
+    def run_status(self):
+        if self._run_status is None:
+            self._run_status = rest_communication.get_document('lims/run_status', match={'run_id': self.name})
+            if not self._run_status:
+                raise AnalysisDriverError('Run status information for %s is not available in the LIMS' % self.name)
+        return self._run_status
+
+    def _run_elements_from_lims_endpoint(self):
         run_elements = []
 
-        flowcell = set(self.lims_run.parent_processes()).pop().output_containers()[0]
-        for lane in flowcell.placements:
-            if len(flowcell.placements[lane].reagent_labels) > 1:
-                artifacts = self._find_pooling_step_for_artifact(flowcell.placements[lane], 'Create PDP Pool')
-            else:
-                artifacts = [flowcell.placements[lane]]
-            for artifact in artifacts:
-                assert len(artifact.samples) == 1
-                sample = artifact.samples[0]
+        for lane in self.run_status['lanes']:
+            for sample_data in lane['samples']:
                 run_element = {
-                    ELEMENT_PROJECT_ID: sample.project.name,
-                    ELEMENT_SAMPLE_INTERNAL_ID: sample.name,
-                    ELEMENT_LIBRARY_INTERNAL_ID: sample.id,  # This is not the library id but it is unique
-                    ELEMENT_LANE: lane.split(':')[0],
+                    ELEMENT_PROJECT_ID: sample_data['project_id'],
+                    ELEMENT_SAMPLE_INTERNAL_ID: sample_data['sample_id'],
+                    ELEMENT_LIBRARY_INTERNAL_ID: sample_data['artifact_id'],  # This is not the library id but it is unique
+                    ELEMENT_LANE: str(lane['lane']),
```

Comment:
`lane` is an integer in Reporting-App's schema.yaml

Reply:
True but it seems like it has always been a string at this point because it came from the artifact position in the flowcell `'1:1'` which got parsed.
If I don't cast it here it fail to find the the lane at 
https://github.com/EdinburghGenomics/Analysis-Driver/blob/199cb668fccc63c95bc7e148722076d82948d527/analysis_driver/report_generation/run_crawler.py#L271
I could have cast it there but I prefer to keep backward compatibility to avoid having other issues elsewhere.
The upload to the reporting-app force the cast to `int`